models/clip/extract_clip.py

```diff
@@ -0,0 +1,175 @@
+        # and cap.read returns None but the rest of the frames are ok. timestep is 0.0 for the 2nd frame in
+        # this case
+        first_frame = True
+        while cap.isOpened():
+            frame_exists, rgb = cap.read()
+
+            if first_frame:
+                first_frame = False
+                if frame_exists is False:
+                    continue
+
+            if frame_exists:
+                timestamps_ms.append(cap.get(cv2.CAP_PROP_POS_MSEC))
+                # prepare data (first -- transform, then -- unsqueeze)
+                rgb = cv2.cvtColor(rgb, cv2.COLOR_BGR2RGB)
+                rgb = preprocess(Image.fromarray(rgb))
+                rgb = rgb.unsqueeze(0)
+                batch.append(rgb)
+                # when batch is formed to inference
+                if len(batch) == self.batch_size:
+                    _run_on_a_batch(vid_feats, batch, model, device)
+                    # clean up the batch list
+                    batch = []
+            else:
+                # if the last batch was smaller than the batch size
+                if len(batch) != 0:
+                    _run_on_a_batch(vid_feats, batch, model, device)
+                cap.release()
+                break
+
+        # removes the video with different fps if it was created to preserve disk space
+        if (self.extraction_fps is not None) and (not self.keep_tmp_files):
+            os.remove(video_path)
+
+        features_with_meta = {
+            self.feature_type: np.array(vid_feats),
+            'fps': np.array(fps),
+            'timestamps_ms': np.array(timestamps_ms)
+        }
+
+        return features_with_meta
+
+    def load_model(self, device: torch.device) -> Tuple[torch.nn.Module, Callable]:
+        '''Defines the models, loads checkpoints, sends them to the device.
+
+        Args:
+            device (torch.device): The device
+
+        Raises:
+            NotImplementedError: if flow type is not implemented.
+
+        Returns:
+            Tuple[torch.nn.Module, Callable]: the model and the transform function
+        '''
+        # ['RN50', 'RN101', 'RN50x4', 'RN50x16', 'ViT-B/32', 'ViT-B/16']
+        if self.feature_type == 'CLIP-ViT-B-32':
+            model, preprocess = clip.load("ViT-B/32", device=device)
+        elif self.feature_type == 'CLIP-ViT-B-16':
+            model, preprocess = clip.load("ViT-B/16", device=device)
+        elif self.feature_type == 'CLIP-RN50x16':
+            model, preprocess = clip.load("RN50x16", device=device)
+        elif self.feature_type == 'CLIP-RN50x4':
+            model, preprocess = clip.load("RN50x4", device=device)
+        elif self.feature_type == 'CLIP-RN101':
+            model, preprocess = clip.load("RN101", device=device)
+        elif self.feature_type == 'CLIP-RN50':
+            model, preprocess = clip.load("RN50", device=device)
+        elif self.feature_type == 'CLIP-custom':
```

Comment:
Let's remove the custom model support for now :/

Reply:
I will use this feature in my research, but it doesn't matter.

Reply:
Oh, I see. Actually, that's ok. Let's leave it. I just realized that you put the doc already! Great.